### ISSUE TITLE
Saving a tx sending "max" with a set of utxo will send entire balance when selection is cleared. [Possible lost of funds]

### ISSUE THREAD

**Issue by Transisto** [spesmilo/electrum#7245]
Possible solution, 
Dont allow saving MAX when a UTXO selection was made.

**Comment by ecdsa** [comment]
the issue is that we always save an "invoice" when an on-chain transaction is initiated.
maybe we should save the current max amount instead of 'max'

**Comment by SomberNight** [comment]
Personally I use this feature quite often.
How else would you spend some UTXOs without creating a change output? Not just re the Send tab but also e.g. opening a channel: it's quite nice not to create change.

One frequent use case that I have for coin selection is simulating/testing how a wallet would behave if it only had the coins I have selected (this allows quickly testing even e.g. an empty wallet). Saving/paying an invoice paying "max" is part of this.

Also, the saved invoices list shows "max",
and then if you try to pay the invoice, it will explicitly show the amount to be sent and you need to enter your password and confirm.
I think this is fine as-is.
![pic1a](https://user-images.githubusercontent.com/29142493/116080433-96c22100-a688-11eb-96e5-5aa0b91b7681.png)
![pic2a](https://user-images.githubusercontent.com/29142493/116080437-97f34e00-a688-11eb-8a9c-49427cc99fc2.png)

>  maybe we should save the current max amount instead of 'max'

How exactly would that work? What about fees? How would one pay such a saved invoice without creating change?

----

> Possible solution,
Dont allow saving MAX when a UTXO selection was made.

Ok, and what about frozen coins? Look at my second screenshot, and note the difference between the balance and the amount to be sent (which is "max"). It is due to frozen coins. One could argue we should not save "max" invoices if there are any coins frozen.
I am going to argue the opposite: keep the behaviour as-is. The user is shown exactly what is going to be sent and they need to confirm it before it gets sent.

**Comment by ecdsa** [comment]
I agree, it is probably better to keep it as it is.  if we decided not to save invoices with "max", we would still need to save the invoice with something in it, not clear what that would be.
